### PR TITLE
Make dockerfile use jupyter docker stacks

### PR DESCRIPTION
**Pull request by poplav**
Fixes https://github.com/IRkernel/IRkernel/issues/381:
* Adds `dev_image` and `dev` targets to the makefile
  * `dev_image` - builds/installs necessary IRkernel dependencies from the [jupyter docker-stacks/r-noteook](https://github.com/jupyter/docker-stacks/tree/master/r-notebook) base image
  * `dev` - mounts IRkernel source into the `dev_image` container, installs the the IRkernel package from the source, and starts the Jupyter notebook
* Removes old Dockerfile as it has been replaced.
* Makes the distinction between running a stable release version of the irkernel in docker vs running in dev mode in the readme

### CODE REVIEW

**Comment by flying-sheep** [comment]
i have no idea how docker works, so someone else has to review this

**Comment by takluyver** [comment]
> i have no idea how docker works

Magic! :sparkles: 

I'll take a look

File: Makefile [hunk 1]
```diff
@@ -9,3 +13,19 @@ check:
...
 test: 
 	Rscript  -e "library(testthat); test()"
+
+dev_image:
+	@docker run -it --name irkernel-build \
+	$(DEV_IMAGE) bash -c 'Rscript -e "install.packages(c('\''devtools'\''), repos = c('\''http://irkernel.github.io/'\'', '\''http://cran.rstudio.com'\''))" && \
+	                        Rscript -e "library(devtools)" \
+	                        -e "install_github('\''IRkernel/repr'\'')" \
+	                        -e "install_github('\''IRkernel/IRdisplay'\'')"'
+	@docker commit irkernel-build $(DEV_IMAGE)
```
Comment (takluyver) [inline Makefile]: Shouldn't this be committing with a different name, not overwriting `jupyter/r-notebook`? The user should still be able to run the stable version after they've run this, without the two interfering.

File: Makefile [hunk 2]
```diff
@@ -9,3 +13,19 @@ check:
...
+	@docker rm -f irkernel-build
+
+dev:
```
Comment (takluyver) [inline Makefile]: And shouldn't `dev` depend on `dev-image`?

**Comment by poplav** [comment]
@takluyver, added a distinction between base/dev image and added the dependency.

**Comment by takluyver** [comment]
Thanks, that looks better. I assume you've tested doing `make dev`?

I also think we should rename the Make targets so they're more clearly associated with docker, e.g. `docker_dev` and `docker_dev_image`.

File: Makefile [hunk 3]
```diff
@@ -9,3 +14,19 @@ check:
...
 	Rscript  -e "library(testthat); test()"
+
+docker_dev_image:
+	@docker run -it --name irkernel-build \
+	$(BASE_IMAGE) bash -c 'Rscript -e "install.packages(c('\''devtools'\''), repos = c('\''http://irkernel.github.io/'\'', '\''http://cran.rstudio.com'\''))" && \
+	                        Rscript -e "library(devtools)" \
+	                        -e "install_github('\''IRkernel/repr'\'')" \
+	                        -e "install_github('\''IRkernel/IRdisplay'\'')"'
+	@docker commit irkernel-build $(DEV_IMAGE)
+	@docker rm -f irkernel-build
```
Comment (parente) [inline Makefile]: With modern docker (1.9+?), it's better to contain these steps in a Dockerfile.dev and simply `docker build -f Dockerfile.dev -t $(DEV_IMAGE) .`  to get the added value of caching.

Comment (parente) [inline Makefile]: https://github.com/jupyter-incubator/dashboards/blob/master/Makefile#L21

and

https://github.com/jupyter-incubator/dashboards/blob/master/Dockerfile.dev

for reference.

**Comment by poplav** [comment]
@parente, updated and I did notice the caching.

**Comment by takluyver** [comment]
Thanks!

**Pull request merged**